cfgov/v1/models/sublanding_page.py

```diff
@@ -75,10 +75,8 @@ def get_browsefilterable_posts(self, request, limit):
                 filtered_controls.update({id: []})
             form_class = page.get_form_class()
             posts = page.get_page_set(form_class(parent=page, hostname=request.site.hostname), request.site.hostname)
-            if filtered_controls[id]:
-                filtered_controls[id] += posts
-            else:
-                filtered_controls[id] = posts
-        posts_tuple_list = [(id, post) for id, posts in filtered_controls.iteritems() for post in posts]
+            filtered_controls[id].append(posts)
+
+        posts_tuple_list = [(id, post) for id, posts in filtered_controls.iteritems() for post in chain(*posts)]
```

Comment:
Agreed. The use of a dictionary here and then using `chain` to iterate over a list of lists seems a bit overly complicated. Could this logic be made a bit more straightforward with something like this?

```py
posts = []

for page in filter_pages:
    page_id = str(util.get_form_id(page))

    form_class = page.get_form_class()
    page_posts = page.get_page_set(
        form_class(parent=page, hostname=request.site.hostname),
        request.site.hostname
    )

    posts.extend((page_id, page_post) for page_post in page_posts)

posts_by_most_recent = sorted(posts, key=lambda p: p[1].date_published, reverse=True)

return posts_by_most_recent[:limit]
```